File: python_test/iterator.py

class   Iterator:

    def __iter__(self):
        self.var=0
        return self

    def __next__(self):
        self.var += 1
        return self.var

aList = [1, 3, "Hello, World!"]

listIter = iter(aList)

it = iter(Iterator())

print(next(it))
print(next(it))
print(next(it))


# This exactly how [ for element in iterable ] works


#print(next(listIter))
#print(next(listIter))

# while True:

 #   print(listIter)

 #   try:
 #       next(listIter);

 #   except StopIterator:
 #       break

File: python_test/iterator.cpp
# include <iostream>
# include <vector>

int	main(void)
{
	std::vector<int>	v;


	v.push_back(3);
	v.push_back(4);
	v.push_back(5);

	std::cout << "size: " << v.size() << std::endl;	
	std::vector<int>::iterator iter = v.begin();
	while (iter != v.end())
	{
		std::cout << "addr: " << &iter << std::endl;
		std::cout << "value: " << *iter << std::endl; 
		iter++;
	}
	return (EXIT_SUCCESS);
}
